### PR TITLE
docs: image auth design

### PR DESCRIPTION
**Pull request by Xynnn007**
Backgrounds, Goals, Work process and considerations of image auth.
This is linked to https://github.com/confidential-containers/image-rs/issues/63.

cc @stevenhorsman @liudalibj

### CODE REVIEW

**Comment by liudalibj** [comment]
@Xynnn007 The design looks good to me, thanks for your quick action!

**Review by liudalibj — APPROVED** [review]
LGTM

**Review by arronwy — APPROVED** [review]
Thanks @Xynnn007 , LGTM!

File: docs/image_auth.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+## Considerations
+
+### Where to store the `auth.json`?
+
+The same as `policy.json` in [image security](ccv1_image_security_design.md#policy).
+We put the file in `/run/image-security/auth.json` inside the Pod, because the `/run` directory is mounted in `tmpfs`, which is located in the encrypted memory protected by HW-TEE.
+
+### When `image-rs` should check and get `auth.json` if not present?
+
+There are two options:
```
Comment (jialez0) [inline docs/image_auth.md:82]: I wonder whether this document is a proposal or a design document. If it is the latter, should there be no multiple options and `may` statements in the document?

Comment (Xynnn007) [reply docs/image_auth.md:82]: Thanks, this document is a design document. I'll go to replace `may` statements. Besides, the reason why put options here is that this is not discussed before, and rationale of the choice should be declared.

**Review by jialez0 — APPROVED** [review]
LGTM.

**Pull request merged**